### PR TITLE
Update-eol-versions-6.0-staging

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

**Review by carlossanlop — APPROVED** [review]
@richlander thanks for submitting, this looks good.

Question: I am seeing some 6.0 PRs failing with this error too:
```
SENDHELIXJOB(0,0): error : Helix queue redhat.7.amd64.open is set for estimated removal date of 2024-06-30. In most cases the queue will be removed permanently due to end-of-life; please contact dnceng for any questions or concerns, and we can help you decide how to proceed and discuss other options.
```
Do you know what's the now supported version we should be using? This does not need to be addressed in this PR, can be done separately.

Aside from the above question, is this ready to merge? If yes, feel free to press the squash button.

**Comment by richlander** [comment]
Libraries tests are failing on Windows 10.

```bash
    System.Net.Http.WinHttpHandlerFunctional.Tests.BidirectionStreamingTest.BackwardsCompatibility_DowngradeToHttp11 [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.Exception)
      Actual:   (No exception was thrown)
      Stack Trace:
        /_/src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/BidirectionStreamingTest.cs(335,0): at System.Net.Http.WinHttpHandlerFunctional.Tests.BidirectionStreamingTest.BackwardsCompatibility_DowngradeToHttp11()
        --- End of stack trace from previous location ---
```

@wfurt

**Comment by richlander** [comment]
Looks like something is wrong with either the infra or `System.IO.Net5Compat.Tests` and `System.IO.Tests' (or the underlying product areas).

@carlossanlop

**Comment by carlossanlop** [comment]
/ba-g System.IO and System.IO.Net5Compat failure is a known/pre-existing issue that KnownBuildError was unable to link: https://github.com/dotnet/runtime/issues/100558

File: eng/pipelines/coreclr/templates/helix-queues-setup.yml [hunk 1]
```diff
@@ -36,68 +36,68 @@ jobs:
...
       - ${{ if eq(variables['System.TeamProject'], 'public') }}:
-        - (Ubuntu.1804.Arm32.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7
+        - (Ubuntu.1804.Arm32.Open)Ubuntu.2204.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7
       - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
-        - (Debian.10.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-arm32v7
-        - (Debian.11.Arm32)Ubuntu.1804.ArmArch@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-arm32v7
-        - (Ubuntu.1804.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7
+        - (Debian.12.Arm32)Ubuntu.2204.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-12-helix-arm32v7
+        - (Debian.11.Arm32)Ubuntu.2204.ArmArch@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-arm32v7
+        - (Ubuntu.2004.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7
```
Comment (akoeplinger) [inline eng/pipelines/coreclr/templates/helix-queues-setup.yml:52]: this is not correct, the value in parens is just a display name. the actual docker image is still using 18.04

Comment (richlander) [reply eng/pipelines/coreclr/templates/helix-queues-setup.yml:52]: Thanks for raising this and the other one. I'll take a look at both later today and fix the mistakes.

Comment (richlander) [reply eng/pipelines/coreclr/templates/helix-queues-setup.yml:52]: #105063

**Pull request merged**